### ISSUE TITLE
Test hangs when output message to stdout that length is larger than 3743 

### ISSUE THREAD

**Issue by filzrev** [xunit/visualstudio.xunit#428]
When running test with xUnit v3 and Test Explorer using Visual Studio 2022 **Version 17.13.0 Preview 2.0**.
Tests are executed by using [Microsoft.Testing.Platform protocol](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-vs-vstest#communication-protocol-preview).

And If test output messages that total length is larger than `3743`.
Running test don't completed on Test Explorer, (When running tests with `dotnet test`. It successfully completed)

**Test Code**

```
 [Fact]
 public void Test1()
 {
     var message = new string('-', 3743); // Pass without problems
     Console.Write(message);
 }

 [Fact]
 public void Test2()
 {
     var message = new string('-', 3744); // Hanged when running test with Visual Studio Preview TestExplorer.
     Console.Write(message);
 }
```

**Additional information**

When using `[assembly: CaptureConsole]` assembly level attribute.
Test passed as expected.  But in this case, console outputs are captured
So I want another options to discard stdout/stderr automatically.

I'm also try to set flag to disable [Microsoft.Testing.Platform protocol](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-vs-vstest#disabling-the-new-protocol)
But it seems not works.

**Comment by stevendarby** [comment]
Experiencing the same probme. Thankfully I found this issue with the explanation, so thanks @filzrev. In my case I was using EF Core in tests and had configured the `DbContextOptions` with `LogTo(Console.WriteLine)`. Removing this gets the tests running again for now.

**Comment by bradwilson** [comment]
Given that this does not repro anywhere except for Test Explorer, and only in the preview version of Visual Studio, I'm guessing the bug should be reported against Test Explorer. I'm unable to reproduce this at all with VS 2022 127.12.3, nor with `dotnet test`, `dotnet run`, or `xunit.v3.runner.console`.

**Comment by filzrev** [comment]
> Given that this does not repro anywhere except for Test Explorer, and only in the preview version of Visual Studio

I've confirmed it can be reproduced on following lates Visual Studio versions.

- Visual Studio 2022 Version `17.12.3` (With preview feature `Use testing platform server mode` enabled)
- Visual Studio 2022 Version `17.13.0 Preview 2.1`

In this case. error threshold value is changed from `3744` to `3741`.

> I'm guessing the bug should be reported against Test Explorer. 

I'm suspecting it cause deadlock problems due to buffer full in pipe that is used to cross-process communications.
If it can able to distinguish root cause of problems.
I would like to create ticket on the `testfx` repository or `https://developercommunity.visualstudio.com`

**Current workaround**
- Use `[assembly: CaptureConsole]` to capture stout
- Explicitly discard `stdout` by temporary setting `Console.SetError(TextWriter.Null)`

**Comment by bradwilson** [comment]
> Visual Studio 2022 Version 17.12.3 (With preview feature Use testing platform server mode enabled)

My point remains that this appears to be a preview Test Explorer bug, not a bug in xUnit.net.

**Comment by bradwilson** [comment]
> Use [assembly: CaptureConsole] to capture stout

If the problem is from StdErr and not StdOut, you can tell `[CaptureConsole]` to only capture StdOut.

**Comment by stevendarby** [comment]
In case it's of any interest, it doesn't repro on the VS Preview versions when using latest v2, it's something specifically about the combination of the preview Test Explorer and xunit v3. Happy to raise this on the Test Explorer side though :)

**Comment by bradwilson** [comment]
Right, v2 doesn't support Microsoft Testing Platform, so that's going to be a factor in the bug.

**Comment by bradwilson** [comment]
In essence, Test Explorer is unchanged for non-MTP test projects. The preview for supporting MTP creates an entirely new behavioral path chain. If my guess about origin is correct, you will be able to repro this using MSTest 3.6+ with MTP enabled as well.

**Comment by bradwilson** [comment]
Hello,

As part of the effort to clean up older issues, this is being closed due to being designated an external issue.

Thanks!

**Comment by filzrev** [comment]
This issue is reported on `Developer Community` and tracked on follwoing URL.

https://developercommunity.visualstudio.com/t/Test-Explorer-failed-to-run-xUnit-v3-te/10819610

**Comment by filzrev** [comment]
This issue seems to be fixed in the latest VS versions.
I've confirmed with following versions.
 - v17.13.0 (Note: MTP feature is enabled by default from this version)
 - v17.14.0 Preview 1.0

**Comment by mdh-origami** [comment]
For reference for other people, in my situation I was using a WebApplicationFactory to create a client and just doing that with XUnit.v3 and Visual Studio 17.12.X hung when running the test normally, but not when debugging.  When I upgraded to VS 17.13.X it runs fine.
